src/commands/distribute/release.ts

```diff
@@ -0,0 +1,224 @@
+    }
+  }
+
+  private getPrerequisites(client: MobileCenterClient): Promise<[models.DistributionGroupUserGetResponse[], Buffer, string]> {
+    // load release binary file
+    const fileBuffer = Pfs.readFile(this.filePath).catch((error: NodeJS.ErrnoException) => {
+      if (error.code === "ENOENT") {
+        throw failure(ErrorCodes.InvalidParameter, `binary file '${this.filePath}' doesn't exist`);
+      } else {
+        throw error;
+      }
+    });
+
+    // load release notes file or use provided release notes if none was specified
+    const releaseNotesString = !_.isNil(this.releaseNotesFile) ? 
+      Pfs.readFile(this.releaseNotesFile, "utf8").catch((error: NodeJS.ErrnoException) => {
+        if (error.code === "ENOENT") {
+          throw failure(ErrorCodes.InvalidParameter, `release notes file '${this.releaseNotesFile}' doesn't exist`);
+        } else {
+          throw error;
+        }
+      })
+     : Promise.resolve(this.releaseNotes);
+    
+    // get distribution group users (and check distribution group existence)
+    const distributionGroupUsers = this.getDistributionGroupUsers(client);
+
+    return Promise.all([distributionGroupUsers, fileBuffer, releaseNotesString]);
+  }
+
+  private async getDistributionGroupUsers(client: MobileCenterClient): Promise<models.DistributionGroupUserGetResponse[]> {
+    const distributionGroupUsersRequestResponse = await clientRequest<models.DistributionGroupUserGetResponse[]>(
+      (cb) => client.account.getDistributionGroupUsers(this.app.ownerName, this.app.appName, this.distributionGroup, cb));
+    const statusCode = distributionGroupUsersRequestResponse.response.statusCode;
+    if (statusCode >= 400) {
+      if (statusCode === 404) {
+        throw failure(ErrorCodes.InvalidParameter, `distribution group ${this.distributionGroup} was not found`);
+      } else {
+        throw failure(ErrorCodes.Exception, `failed to get users of distribution group ${this.distributionGroup}`);
+      }
+    } else {
+      return distributionGroupUsersRequestResponse.result;
+    }
+  }
+
+  private async createReleaseUpload(client: MobileCenterClient, app: DefaultApp): Promise<models.ReleaseUploadBeginResponse> {
+    const createReleaseUploadRequestResponse = await out.progress("Creating release upload...", 
+      clientRequest<models.ReleaseUploadBeginResponse>((cb) => client.distribute.createReleaseUpload(app.ownerName, app.appName, cb)));
+
+    if (createReleaseUploadRequestResponse.response.statusCode >= 400) {
+      throw failure(ErrorCodes.Exception, `failed to create release upload for ${this.filePath}`);
+    } else {
+      return createReleaseUploadRequestResponse.result;
+    }
+  }
+
+  private uploadFileToUri(uploadUrl: string, fileBuffer: Buffer, filename: string): Promise<void> {
+    debug("Uploading the release binary");
+    return new Promise<void>((resolve, reject) => {
+      Request.post({
```

Comment:
We're looking at adding the `node-fetch` module, which will give us a promise-native API for doing raw web requests Until then, this is fine.